### PR TITLE
For #27117 - update copy in the menu from sign in to sync and save

### LINKED ISSUE #27117: Update copy for the menu from "Sign in to sync" to "Sync and save data"
Update the Firefox menu to read `Sync and save data`. This aligns wording with desktop and iOS.

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FNXV2-21635)

### PR DESCRIPTION
**Pull request by skhamis**
This should line up with https://phabricator.services.mozilla.com/D159227 so they land around the same time for string consistency.

Screen shot of new copy change:
Screen Shot 2022-10-13 at 9 23 21 AM

### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### QA

- [ ] **QA Needed**

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-debug` task.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
Fixes #27117

### CODE REVIEW

**Comment by MozillaNoah** [comment]
The changes look good! Before we land this, could you please squash the two commits into one and update the commit message to be prefixed with `For #27117 - ...`

File: app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -1042,10 +1042,10 @@
     <string name="share_copy_link_to_clipboard">Copy to clipboard</string>
     <!-- Toast shown after copying link to clipboard -->
     <string name="toast_copy_link_to_clipboard">Copied to clipboard</string>
-    <!-- An option from the three dot menu to into sync -->
-    <string name="sync_menu_sign_in">Sign in to sync</string>
```
Comment (gabrielluong) [inline app/src/main/res/values/strings.xml:1046]: `<string name="sync_menu_sign_in" moz:removedIn="107" tools:ignore="UnusedResources">Sign in to sync</string>`

Unfortunately, we can't simply remove a string. Please see https://github.com/mozilla-mobile/fenix/wiki/Removing-strings#marking-an-unused-string-to-be-removed

File: app/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -1042,10 +1042,10 @@
...
     <!-- An option from the share dialog to sign into sync -->
     <string name="sync_sign_in">Sign in to Sync</string>
+     <!-- An option from the three dot menu to sync and save data -->
+    <string name="sync_menu_and_save_data">Sync and save data</string>
```
Comment (gabrielluong) [inline app/src/main/res/values/strings.xml:1048]:
```suggestion
    <string name="sync_menu_sync_and_save_data">Sync and save data</string>
```

File: app/src/androidTest/java/org/mozilla/fenix/ui/robots/ThreeDotMenuMainRobot.kt [hunk 3]
```diff
@@ -173,7 +173,7 @@ class ThreeDotMenuMainRobot {
 
         fun openSyncSignIn(interact: SyncSignInRobot.() -> Unit): SyncSignInRobot.Transition {
             threeDotMenuRecyclerView().perform(swipeDown())
-            mDevice.waitNotNull(Until.findObject(By.text("Sign in to sync")), waitingTime)
+            mDevice.waitNotNull(Until.findObject(By.text("Save and sync data")), waitingTime)
```
Comment (gabrielluong) [inline app/src/androidTest/java/org/mozilla/fenix/ui/robots/ThreeDotMenuMainRobot.kt:176]:
```suggestion
            mDevice.waitNotNull(Until.findObject(By.text("Sync and save data")), waitingTime)
```

**Comment by gabrielluong** [comment]
Could we also update the commit message to follow the original issue title - Update copy for the menu from "Sign in to sync" to "Sync and save data"?

**Review by gabrielluong — APPROVED** [review]

**Pull request merged**